x/stake/client/rest/utils.go

```diff
@@ -0,0 +1,205 @@
+			return types.UnbondingDelegation{}, http.StatusInternalServerError, "couldn't unmarshall unbonding-delegation. Error: ", errUnmarshal
+		}
+		return unbondingDelegation, http.StatusOK, "", nil
+	}
+	return types.UnbondingDelegation{}, http.StatusNoContent, "", nil
+}
+
+func getDelegatorRedelegations(ctx context.CoreContext, cdc *wire.Codec, delegatorAddr sdk.AccAddress, validatorAddr sdk.AccAddress) (
+	regelegations types.Redelegation, httpStatusCode int, errMsg string, err error) {
+
+	keyRedelegateTo := stake.GetREDsByDelToValDstIndexKey(delegatorAddr, validatorAddr)
+	marshalledRedelegations, err := ctx.QueryStore(keyRedelegateTo, storeName)
+	if err != nil {
+		return types.Redelegation{}, http.StatusInternalServerError, "couldn't query redelegation. Error: ", err
+	}
+
+	if len(marshalledRedelegations) != 0 {
+		redelegations, errUnmarshal := types.UnmarshalRED(cdc, keyRedelegateTo, marshalledRedelegations)
+		if errUnmarshal != nil {
+			return types.Redelegation{}, http.StatusInternalServerError, "couldn't unmarshall redelegations. Error: ", errUnmarshal
+		}
+
+		return redelegations, http.StatusOK, "", nil
+	}
+	return types.Redelegation{}, http.StatusNoContent, "", nil
+}
+
+// queryTxs Queries staking txs
+func queryTxs(node rpcclient.Client, cdc *wire.Codec, tag string, delegatorAddr string) ([]tx.Info, error) {
+	page := 0
+	perPage := 100
+	prove := false
+	query := fmt.Sprintf("%s='%s' AND %s='%s'", tags.Action, tag, tags.Delegator, delegatorAddr)
+	res, err := node.TxSearch(query, prove, page, perPage)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx.FormatTxResults(cdc, res.Txs)
+}
+
+// getValidators Gets all Validators
+func getValidators(validatorKVs []sdk.KVPair, cdc *wire.Codec) ([]types.BechValidator, error) {
+	validators := make([]types.BechValidator, len(validatorKVs))
+	for i, kv := range validatorKVs {
+
+		addr := kv.Key[1:]
+		validator, err := types.UnmarshalValidator(cdc, addr, kv.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		bech32Validator, err := validator.Bech32Validator()
+		if err != nil {
+			return nil, err
+		}
+		validators[i] = bech32Validator
+	}
+	return validators, nil
+}
+
+// getValidator Gets a validator given a ValAddress
+func getValidator(address sdk.ValAddress, validatorKVs []sdk.KVPair, cdc *wire.Codec) (stake.BechValidator, error) {
+	// parse out the validators
+	for _, kv := range validatorKVs {
+		addr := kv.Key[1:]
+		validator, err := types.UnmarshalValidator(cdc, addr, kv.Value)
+		if err != nil {
+			return stake.BechValidator{}, err
+		}
+
+		ownerAddress := validator.PubKey.Address()
+		if bytes.Equal(ownerAddress.Bytes(), address.Bytes()) {
+			bech32Validator, err := validator.Bech32Validator()
+			if err != nil {
+				return stake.BechValidator{}, err
+			}
+
+			return bech32Validator, nil
+		}
+	}
+	return stake.BechValidator{}, errors.Errorf("Couldn't find validator")
+}
+
+// getValidatorFromAccAdrr Gets a validator given an AccAddress
```

Comment:
Tendermint Lint doesn't require that the comments begin with the name of the function... probably preferable to not have them here but doesn't really matter I guess

Reply:
ohh wasn't aware of that I was using go-lint so it kept giving me warnings